docs/architecture/adr-012-peer-transport.md

```diff
@@ -0,0 +1,110 @@
+# ADR 012: PeerTransport
+
+## Context
+
+One of the more apparent problems with the current architecture in the p2p
+package is that there is no clear separation of concerns between different
+components. Most notably the `Switch` is currently doing physical connection
+handling. An artifact is the dependency of the Switch on `config.P2PConfig`.
```

Comment:
Link would be helpful again.

Reply:
Link to what in that section would you like to see?

Reply:
[`config.P2PConfig`](https://github.com/tendermint/tendermint/blob/05a76fb517f50da27b4bfcdc7b4cf185fc61eff6/config/config.go#L272-L339)